### PR TITLE
Project Topping with additional Layers

### LINKED ISSUE #593: Loading of additional layer tree structures from configuration file
It should be possible to load an additional layer tree structure (defined in a .qlr file).

This should happen before additional relations are loaded ( see also issue #589 )

As a configuration parameter the following should be possible:
- top (before any other layer)
- bottom (below any other layer)
- beforeLayer(layer_name)
- afterLayer(layer_name)

Not sure what configuration file would be best (open to discussion):
- additional entries in the layertree .yaml ?
- additional entries in the metaconfig .ini file?

Additional issue/discussion: what should happen with the layer order? Can we have parameters that help us arrange layer order as well?

### PR DESCRIPTION
**Pull request by signedav**
This PR allowes to define additional layers in the layertree file by adding the QLR path/link or the data source. 

This resolves #593

Additionally it's possible (and suggested) to define the QML path/link directly in the layertree file as well.

And to define a layer's source directly. Currently supported provider types are `ogr`, `postgres` and `wms`.

## Example

```yaml
layertree:
    - "KbS_LV95_V1_4 Layers":
        group: true
        child-nodes:
            - "Belasteter_Standort (Geo_Lage_Punkt)":
                qmlstylefile: "ilidata:ch.opengis.topping.opengisch_KbS_LV95_V1_4_004"
            - "Parzellenidentifikation":
                qmlstylefile: "ilidata:ch.opengis.topping.opengisch_KbS_LV95_V1_4_005"
    - "Other Layers":
        group: true
        child-nodes:
            - "Roads from QLR": # layer called "The Road Signs" with qlr source
                definitionfile: "../qlr/the_road_signs.qlr"
            - "QLR-Group":# group called "Simple Roads" with qlr source
                group: true
                definitionfile: "../qlr/simple_roads_complete.qlr"
            - "Local WMS": #direct source definition for WMS
                provider: "wms"
                uri: "contextualWMSLegend=0&crs=EPSG:2056&dpiMode=7&featureCount=10&format=image/jpeg&layers=ch.bav.kataster-belasteter-standorte-oev_lines&styles=default&url=https://wms.geo.admin.ch/?%0ASERVICE%3DWMS%0A%26VERSION%3D1.3.0%0A%26REQUEST%3DGetCapabilities"
```

!!! Warning
		The datasource in the QLR file is relative. 

## Changes to the concept

With this functionality it made sense to change the concept. ***Still it's fully backwards compatible with the old keywords etc.***

With the project topping file we have the QGIS project part there, while the ili2db settings / Model Baker settings / data part is in the metaconfig still. It's nicer to have everything at one place the linking in the layer tree is more readable and understandable and a layer is unique considering it's name and position.

### 1. Renaming of the layertree to projecttopping

The layertree is not called layertree anymore since it contains more information (an will in future) than just the legend. 

For that in the metaconfiguration file you will now have instead of `qgis.modelbaker.layertree` the parameter:
```ini
[CONFIGURATION]
qgis.modelbaker.projecttopping=ilidata:ch.opengis.config.KbS_LV95_V1_4_projecttopping
```

In the ilidata.xml there is a new type `http://codes.interlis.ch/type/projecttopping` replacing `http://codes.interlis.ch/type/layertree`.

And inside the projecttopping file (YAML) you will have the section `layertree:` instead of `legend:` since it's more than a legend now.

### 2. Definition of the QML files in the project topping file (YAML)

Previously we defined the mapping between layernames and stylefiles in the metaconfiguration like this:
```ini
[qgis.modelbaker.qml]
"Belasteter_Standort"=ilidata:ch.opengis.topping.opengisch_KbS_LV95_V1_4_001
```

Now best practice is to define the qml files in the layertree. Like in the above example.

### How we create the legend/layers now

First the layers from the physical database are created. They are part of the project in any case.
On creating the legend (with passing the project topping file's layertree (`layertree:` or legacy: `legend:`)

### The path resolver
The project_topping is written in a way of a standard and as well the library part of Model Baker. Means it could be used even without having UsabILIty Hub by passing an own path resolver function. In the Model Baker Plugin case, the passed path resolver is recognizing "ilidata:" prefixed ids and getting it from the UsabILIty Hub. 

The default path_resolver in the library uses the given path (absolute or relative to the python file location). Model Baker Plugin recognizes the ilidata-prefix or uses the path relative to the project topping file.

## To Do:
- [x] Fix tests
- [x] QLR groups are added at the end
- [x] Why are the testfile in the `/24/` directory
- [x] Documentation
- Maybe this #649 
- [x] Maybe that #650
- And check if we need this #479
- [x] Update docs on usability website - see https://github.com/opengisch/usabilityhub/pull/11

### CODE REVIEW

**Comment by signedav** [comment]
Hey @domi4484 Another big PR for you, but don't worry, it has a lot of renamings and the code part is not that huge.

Please have a look at the  py files in libprojgenerator and the wizard. And maybe if the description of the new possibility in docs/background_info/usabilityhub/modelbaker_integration.md is understandable.

Thanks

**Review by domi4484 — APPROVED** [review]
Looks good to me! I have seen just few very minor things

File: QgisModelBaker/gui/generate_project.py [hunk 1]
```diff
@@ -458,48 +458,66 @@ def accepted(self, edited_command=None):
...
+            # Project topping file for legend and layers: collect and download
+            projecttopping_file_path_list = []
             if "CONFIGURATION" in self.metaconfig.sections():
                 configuration_section = self.metaconfig["CONFIGURATION"]
-                if "qgis.modelbaker.layertree" in configuration_section:
+                # get topping referenced in qgis.modelbaker.projecttopping
+                key = "qgis.modelbaker.projecttopping"
+                if key not in configuration_section:
+                    key = "qgis.modelbaker.layertree"
+                if key in configuration_section:
```
[the remark this answers is not in the document]
Comment (signedav) [reply QgisModelBaker/gui/generate_project.py:470]: Good idea.

File: QgisModelBaker/gui/workflow_wizard/project_creation_page.py [hunk 2]
```diff
@@ -247,3 +268,9 @@ def _create_project(self):
 
         self.progress_bar.setValue(100)
         self.workflow_wizard.log_panel.print_info(self.tr("It's served!"))
+
+    def ilidata_path_resolver(self, base_path, path):
+        if "ilidata:" in path or "file:" in path:
+            data_file_path_list = self.workflow_wizard.get_topping_file_list([path])
+            return data_file_path_list[0] if data_file_path_list else None
+        return os.path.join(base_path, path)
```
[the remark this answers is not in the document]
Comment (signedav) [reply QgisModelBaker/gui/workflow_wizard/project_creation_page.py:276]: In the next release we will remove the buttons to the single dialogs (like generate_project.py) and the code will be obsolete then.

**Pull request merged**